crates/pathfinder/src/sequencer.rs

```diff
@@ -121,37 +121,54 @@ async fn parse_raw(resp: reqwest::Response) -> Result<reqwest::Response, Sequenc
 
 /// Wrapper function to allow retrying sequencer queries in an exponential manner.
 ///
-/// Initial backoff time is 2 seconds. Retrying stops after approximately 4 minutes in total.
+/// Retry is performed on __all__ types of errors __except for__
+/// [StarkNet specific errors](crate::sequencer::error::StarknetError).
+///
+/// Initial backoff time is 30 seconds and saturates at 1 hour:
+///
+/// `backoff [secs] = min((2 ^ N) * 15, 3600) [secs]`
+///
+/// where `N` is the consecutive retry iteration number `{1, 2, ...}`.
 async fn retry<T, Fut, FutureFactory>(future_factory: FutureFactory) -> Result<T, SequencerError>
 where
     Fut: Future<Output = Result<T, SequencerError>>,
     FutureFactory: FnMut() -> Fut,
 {
     use crate::retry::Retry;
     use reqwest::StatusCode;
-    use std::num::{NonZeroU64, NonZeroUsize};
+    use std::num::NonZeroU64;
+    use tracing::{error, info, warn};
 
     Retry::exponential(future_factory, NonZeroU64::new(2).unwrap())
-        // Max number of retries of 7 gives a total accumulated timeout of 4 minutes and 15 seconds (2^8-1)
-        .max_num_retries(NonZeroUsize::new(7).unwrap())
+        .factor(NonZeroU64::new(15).unwrap())
+        .max_delay(Duration::from_secs(60 * 60))
         .when(|e| match e {
-            SequencerError::TransportError(te) if te.is_timeout() => {
-                tracing::debug!("Retrying due to timeout");
+            SequencerError::ReqwestError(e) => {
+                if e.is_body() || e.is_connect() || e.is_timeout() {
+                    info!(error=%e, "Request failed, retrying:");
+                } else if e.is_status() {
+                    match e.status() {
+                        Some(StatusCode::NOT_FOUND | StatusCode::TOO_MANY_REQUESTS | StatusCode::BAD_GATEWAY
+                            | StatusCode::SERVICE_UNAVAILABLE
+                            | StatusCode::GATEWAY_TIMEOUT) => {
+                            info!(error=%e, "Request failed, retrying:");
+                        }
+                        Some(StatusCode::INTERNAL_SERVER_ERROR) => {
+                            unreachable!("All HTTP 500s should either be converted to StarkNet errors or result in a deserialization error.")
```

Comment:
I think maybe lets just make this a `error!` as well; so that we don't crash but still know there's a bug?